pkg/sql/sqlbase/metadata.go

```diff
@@ -149,36 +147,26 @@ func (ms MetadataSchema) GetInitialValues(
 		// Create name metadata key.
 		value := roachpb.Value{}
 		value.SetInt(int64(desc.GetID()))
-
-		// TODO(solon): This if/else can be removed in 20.2, as there will be no
-		// need to support the deprecated namespace table.
-		if bootstrapVersion.IsActive(clusterversion.VersionNamespaceTableWithSchemas) {
-			if parentID != keys.RootNamespaceID {
-				ret = append(ret, roachpb.KeyValue{
-					Key:   NewPublicTableKey(parentID, desc.GetName()).Key(ms.codec),
-					Value: value,
-				})
-			} else {
-				// Initializing a database. Databases must be initialized with
-				// the public schema, as all tables are scoped under the public schema.
-				publicSchemaValue := roachpb.Value{}
-				publicSchemaValue.SetInt(int64(keys.PublicSchemaID))
-				ret = append(
-					ret,
-					roachpb.KeyValue{
-						Key:   NewDatabaseKey(desc.GetName()).Key(ms.codec),
-						Value: value,
-					},
-					roachpb.KeyValue{
-						Key:   NewPublicSchemaKey(desc.GetID()).Key(ms.codec),
-						Value: publicSchemaValue,
-					})
-			}
-		} else {
+		if parentID != keys.RootNamespaceID {
 			ret = append(ret, roachpb.KeyValue{
-				Key:   NewDeprecatedTableKey(parentID, desc.GetName()).Key(ms.codec),
+				Key:   NewPublicTableKey(parentID, desc.GetName()).Key(ms.codec),
 				Value: value,
 			})
+		} else {
+			// Initializing a database. Databases must be initialized with
+			// the public schema, as all tables are scoped under the public schema.
+			publicSchemaValue := roachpb.Value{}
+			publicSchemaValue.SetInt(int64(keys.PublicSchemaID))
+			ret = append(
+				ret,
+				roachpb.KeyValue{
+					Key:   NewDatabaseKey(desc.GetName()).Key(ms.codec),
+					Value: value,
+				},
+				roachpb.KeyValue{
+					Key:   NewPublicSchemaKey(desc.GetID()).Key(ms.codec),
+					Value: publicSchemaValue,
+				})
```

Comment:
I thought this could be problematic given #49092, but I'm pretty sure it's not. This change simply edits what *new* clusters have as initial values. New clusters don't need anything for the old namespace table.

Does that square with your understanding @nvanbenschoten?

Reply:
Yes, that's correct. This is what new clusters use for their initial values.